larray/core/array.py

```diff
@@ -7497,7 +7497,10 @@ def growth_rate(self, axis=-1, d=1, label='upper'):
             array = self
             axis = array.axes[axis]
         diff = array.diff(axis=axis, d=d, label=label)
-        return diff / array[axis.i[:-d]].ignore_labels(axis)
+        res = diff / array[axis.i[:-d]].ignore_labels(axis)
+        # handles the case where consecutive zeros in the original array makes growth_rate() to return NaNs
+        res.data = np.where(diff.data != 0, res.data, 0)
```

Comment:
1. To avoid recursive imports, I have to import the `where()` function from the ufuncs.py module inside the growth_rate method.
2. Even if I do that, running the doctests returns errors. In short, the `where()` function returns a Numpy array (like if the `broadcastify()` function has no impact):
```python
Failed example:
    a.growth_rate()
Expected:
    sex\year  2017  2018  2019  2020
           M   1.0  0.25  -0.2   0.5
           F   0.5  -0.5   1.0   0.5
Got:
    array([[ 1.  ,  0.25, -0.2 ,  0.5 ],
           [ 0.5 , -0.5 ,  1.  ,  0.5 ]])
```
I have no idea such error happens.

Reply:
1. ok
2. Seems very weird. My only explanation is that you somehow ended up importing np.where anyway.

Anyway, my initial negative reaction is that I feared broadcasting between the arguments wouldn't happen properly by using the numpy function instead of ours... But here, given how the two arrays involved (diff and res) are constructed, they will have the same axes so it's no problem and the numpy function is actually better (it is faster with no negative impact that I can see). I would personally add a comment about this saying that we can do so because the axes are guaranteed to be the same.